packages/valory/skills/abstract_round_abci/test_tools/abci_app.py

```diff
@@ -31,119 +31,71 @@
 
-    def end_block(self) -> Tuple[MagicMock, MagicMock]:
+    def end_block(self) -> None:
         """End block."""
-        return MagicMock(), MagicMock()
 
     def check_payload(self, payload: BaseTxPayload) -> None:
-        """Check payloads of type 'payload_a'."""
+        """Check payload."""
 
     def process_payload(self, payload: BaseTxPayload) -> None:
-        """Process payloads of type 'payload_a'."""
+        """Process payload."""
 
 
-class ConcreteRoundB(AbstractRound):
+class ConcreteRoundA(_ConcreteRound):
     """Dummy instantiation of the AbstractRound class."""
 
-    round_id = "concrete_b"
-    allowed_tx_type = "payload_b"
+    round_id = "concrete_a"
+    allowed_tx_type = "payload_a"
 
-    def end_block(self) -> None:
+    def end_block(self) -> Tuple[MagicMock, MagicMock]:  # type: ignore
         """End block."""
+        return MagicMock(), MagicMock()
 
-    def check_payload(self, payload: BaseTxPayload) -> None:
-        """Check payloads of type 'payload_b'."""
 
-    def process_payload(self, payload: BaseTxPayload) -> None:
-        """Process payloads of type 'payload_b'."""
+class ConcreteRoundB(_ConcreteRound):
+    """Dummy instantiation of the AbstractRound class."""
+
+    round_id = "concrete_b"
+    allowed_tx_type = "payload_b"
 
 
-class ConcreteRoundC(AbstractRound):
+class ConcreteRoundC(_ConcreteRound):
     """Dummy instantiation of the AbstractRound class."""
 
     round_id = "concrete_c"
     allowed_tx_type = "payload_c"
 
-    def end_block(self) -> None:
-        """End block."""
 
-    def check_payload(self, payload: BaseTxPayload) -> None:
-        """Check payloads of type 'payload_c'."""
-
-    def process_payload(self, payload: BaseTxPayload) -> None:
-        """Process payloads of type 'payload_c'."""
-
-
-class ConcreteBackgroundRound(AbstractRound):
+class ConcreteBackgroundRound(_ConcreteRound):
     """Dummy instantiation of the AbstractRound class."""
 
     round_id = "concrete_background"
     allowed_tx_type = "payload_background_c"
 
-    def end_block(self) -> None:
-        """End block."""
 
-    def check_payload(self, payload: BaseTxPayload) -> None:
-        """Check payloads of type 'payload_background_c'."""
-
-    def process_payload(self, payload: BaseTxPayload) -> None:
-        """Process payloads of type 'payload_background_c'."""
-
-
-class ConcreteTerminationRoundA(AbstractRound):
+class ConcreteTerminationRoundA(ConcreteRoundA):
     """Dummy instantiation of the AbstractRound class."""
 
     round_id = "concrete_termination_a"
     allowed_tx_type = "payload_termination_a"
 
-    def end_block(self) -> Tuple[MagicMock, MagicMock]:
-        """End block."""
-        return MagicMock(), MagicMock()
 
-    def check_payload(self, payload: BaseTxPayload) -> None:
-        """Check payloads of type 'payload_termination_a'."""
-
-    def process_payload(self, payload: BaseTxPayload) -> None:
-        """Process payloads of type 'payload_termination_a'."""
```

Comment:
asked the author about the purpose of returning `MagicMock(), MagicMock()`, was informed it was because it was a copy paste of `ConcreteRoundA`. Hence didn't write a test for coverage and refactored instead.